metrics/resource_view_test.go

```diff
@@ -139,6 +139,24 @@ func TestSetFactor(t *testing.T) {
 	}
 }
 
+func TestResourceAsString(t *testing.T) {
```

Comment:
You might want to have a third resource, or at least serialize r2 or r1 to string more than once:

https://golang.org/doc/go1#iteration

With the current code, there's a 50-50 chance of this test passing wit the old code, I believe.

Reply:
Testing, it looks like the following also fails intermittently with the old code:


```golang
	for i := 0; i < 10; i++ {
		a, b := resourceAsString(r1), resourceAsString(r2)
		if a != b {
			t.Errorf("Same resource serialized differently: %q vs %q", a, b)
		}
	}
```